gateway/src/apicast/policy/headers.lua

```diff
@@ -0,0 +1,106 @@
+--- Headers policy
+-- This policy allows to include custom headers that will be sent to the
+-- upstream as well as modify or delete the ones included in the original
+-- request.
+-- Similarly, this policy also allows to add, modify, and delete the headers
+-- included in the response.
+
+local ipairs = ipairs
+
+local policy = require('apicast.policy')
+local _M = policy.new('Headers policy')
+
+local new = _M.new
+
+local function new_header_value(current_value, value)
+  if current_value then
+    return current_value .. ', ' .. value
```

Comment:
So I read https://tools.ietf.org/html/rfc7230#section-3.2.2 and it looks like this is more complex than I though because there is an exception: set-cookie header.

Also this is different on request/response. Request can't have duplicated headers and they have to be sent in one field with commas, response can (and they can be concatenated with comma with some exception).

Also the A.2 of describes Cookie in the request has the same issue. https://arxiv.org/pdf/cs/0105018.pdf?

As it is concatenated with semicolon instead of a comma.

Reply:
>   A sender MUST NOT generate multiple header fields with the same field
   name in a message unless either the entire field value for that
   header field is defined as a comma-separated list [i.e., #(values)]
   or the header field is a well-known exception (as noted below).

>  A recipient MAY combine multiple header fields with the same field
   name into one "field-name: field-value" pair, without changing the
   semantics of the message, by appending each subsequent field value to
   the combined field value in order, separated by a comma.  The order
   in which header fields with the same field name are received is
   therefore significant to the interpretation of the combined field
   value; a proxy MUST NOT change the order of these field values when
   forwarding a message.

A "sender" is the one sending the message (in HTTP request that would be the client, in HTTP response the server). A "recipient" is the party reading ht HTTP message (server in the request, client in the response).

So my understanding is that the message header can be split into several headers with the same name if the value is a comma separated list and it is not on a blacklist (Cookie, Set-Cookie, ...). 


So I think it should be safe to "add" as a second header with the same name. 
Because it is allowed for the sender to send it split as several headers if the header is array-like. We can't maintain a list of array-like headers but we can tell users to figure it out. 

So I think we would be following the RFC if we would "add" a header as:

```
GET /
Header: val
Header: other
```

And then the recipient can concatenate them with comma. And we leave the blacklist on the user. If they do this operation on some unsupported header (like Host) then it is their problem.